vmdb/app/controllers/api_controller/initializer.rb

```diff
@@ -95,23 +106,34 @@ def new_token_mgr(mod, name, api_config)
       # Accessed as @attr_<type>[<name>], much faster than array include?
       #
       def gen_attr_type_hash
-        @attr_time = {}
-        @attr_url  = {}
+        @attr_url       = {}
+        @attr_time      = {}
+        @attr_encrypted = {}
 
-        ATTR_TYPES[:time].each { |attr| @attr_time[attr] = true }
         ATTR_TYPES[:url].each  { |attr| @attr_url[attr]  = true }
-        #
-        # Let's dynamically get the :date and :datetime attributes from the Classes we care about.
-        #
+
+        ATTR_TYPES[:time].each { |attr| @attr_time[attr] = true }
+        gen_time_attr_type_hash
+
+        all_encrypted_attributes.each { |attr| @attr_encrypted[attr] = true }
+      end
+
+      #
+      # Let's dynamically get the :date and :datetime attributes from the Classes we care about.
+      #
+      def gen_time_attr_type_hash
         collection_config.values.each do |cspec|
-          unless cspec[:klass].blank?
-            klass = cspec[:klass].constantize
-            klass.columns_hash.collect  do |name, typeobj|
-              @attr_time[name] = true if %w(date datetime).include?(typeobj.type.to_s)
-            end
+          next if cspec[:klass].blank?
+          klass = cspec[:klass].constantize
+          klass.columns_hash.collect  do |name, typeobj|
+            @attr_time[name] = true if %w(date datetime).include?(typeobj.type.to_s)
```

Comment:
I initially wanted the all encrypted attrs approach, thus the method. Then we went with dynamic fetching this stuff. The default list that I need (default, config, prov hashes) can move to the ATTR_TYPES list. Thanks